pkgs/development/tools/misc/terraform-plugin-docs/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, buildGoModule, fetchFromGitHub, makeWrapper, musl, terraform, go }:
+
+buildGoModule rec {
+  pname = "terraform-plugin-docs";
+  version = "0.16.0";
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = "terraform-plugin-docs";
+    rev = "v${version}";
+    hash = "sha256-5vbi69GMgkzvN3aEQbNTbk99rg+kfvAvUrdDsuyIm9s=";
+  };
+
+  subPackages = [ "cmd/tfplugindocs" ];
+  vendorHash = "sha256-AjW6BokLVDkIWXToJ7wNq/g19xKTAfpQ/gVlKCV5qw0=";
+  allowGoReference = true;
+  nativeBuildInputs = [ musl makeWrapper ];
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.commit=${src.rev}"
+    "-linkmode external"
+    "-extldflags '-static -L${musl}/lib'"
+  ];
```

Comment:
Why did you add musl in nativeBuildInputs and ldflags?

Reply:
This is fairly straight from https://nixos.wiki/wiki/Go to ensure statically compiling works without issues. Seemingly glibc doesn't like static linking as much:

>Linking against glibc.static does not really work because glibc does not really like static linking. You get a warning like warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking. To really create a static build, use musl.

Reply:
@catouc Hello. Thanks for the explanation!

Using musl breaks darwin support, so I think you can rollback the `platforms` line (L44) to:

```nix
    platforms = platforms.linux;
```

What do you think?

Reply:
For now I feel that would be the route, with a future to figure out on if we can add glibc back for darwin support. Did push the change for the support :)